odoo/addons/base/models/ir_ui_view.py

```diff
@@ -1447,3 +1447,109 @@ def _validate_module_views(self, module):
                 view._check_xml()
             except Exception as e:
                 self.raise_view_error("Can't validate view:\n%s" % e, view.id)
+
+    def fix_faulty_views(self, delete=False):
+        """ Use the get_faulty_views and take appropriate actions
+            using disable_if_possible
+            This function is never called but offers the possibility to be called from a server action
+            Please don't remove
+        """
+        for view in self:
+            if not self.view_is_valid(view):
+                faulty_views = self.get_faulty_views(view,view,self.env['ir.ui.view'])
+                # Final validation
+                self._cr.execute('SAVEPOINT end_validation')
+                faulty_views.clean_properly(delete)
+                if self.view_is_valid(view):
+                    self._cr.execute('RELEASE end_validation')
+                else:
+                    self._cr.execute('ROLLBACK TO end_validation')
```

Comment:
`rollback to savepoint` doesn't destroy the savepoint, it still needs to be released. Here this could just be:
```python
if not view.view_is_valid():
    self._cr.execute('ROLLBACK TO end_validation')
    # some sort of error reporting
self._cr.execute('RELEASE end_validation')
```